### PR TITLE
Ensure Config.inifile is available during pytest_cmdline_main

### LINKED ISSUE #9396: [prerelease] Attributes from legacypath plugin added too late?
Based on https://github.com/joeyespo/pytest-watch/issues/122 it seems impossible to access `config.inifile` from `pytest_cmdline_main` now.

Minimal reproducer `conftest.py`:

```python
def pytest_cmdline_main(config):
    print(config.inifile)
```

fails after 84722a7904af41191165dcb5458eb641fd2c4d8b with:

```pytb
[...]
  File ".../conftest.py", line 2, in pytest_cmdline_main
    print(config.inifile)
AttributeError: 'Config' object has no attribute 'inifile'
```

but works fine before.

This is part of [Add legacypath plugin, move py.path stuff there by bluetech · Pull Request #9208 · pytest-dev/pytest](https://github.com/pytest-dev/pytest/pull/9208), cc @bluetech.

Makes me wonder whether this plugin is really a good idea in practice, also seeing things like 663be09723e072be48ffba3fd3f9b4991dcb10c3. Maybe those should be kept in core after all until we can deprecate and remove them, and only then be moved to a (rather hackish) plugin?

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #9396

### CODE REVIEW

File: src/_pytest/legacypath.py [hunk 1]
```diff
@@ -401,26 +401,10 @@ def Node_fspath_set(self: Node, value: LEGACY_PATH) -> None:
 
 
 @pytest.hookimpl
-def pytest_configure(config: pytest.Config) -> None:
+def pytest_load_initial_conftests(early_config: pytest.Config) -> None:
```
Comment (nicoddemus) [inline src/_pytest/legacypath.py:404]: Changing to `pytest_load_initial_conftests` to monkey patch the attributes as it executes earlier than `pytest_configure`.

File: src/_pytest/legacypath.py [hunk 2]
```diff
@@ -452,6 +436,29 @@ def pytest_configure(config: pytest.Config) -> None:
     mp.setattr(Node, "fspath", property(Node_fspath, Node_fspath_set), raising=False)
 
 
+@pytest.hookimpl
+def pytest_configure(config: pytest.Config) -> None:
+    """Installs the LegacyTmpdirPlugin if the ``tmpdir`` plugin is also installed."""
+    if config.pluginmanager.has_plugin("tmpdir"):
```
Comment (nicoddemus) [inline src/_pytest/legacypath.py:442]: The `LegacyTmpdirPlugin` should be installed during `pytest_configure` as that's called after `tmpdir` is installed.

File: changelog/9396.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Ensure :attr:`pytest.Config.inifile` is available during the :func:`pytest_cmdline_main <_pytest.hookspec.pytest_cmdline_main>` hook.
```
Comment (bluetech) [inline changelog/9396.bugfix.rst:1]: Maybe clarify this only fixes a regression in `7.0.0rc1`, i.e. not relevant for most users.

**Review by bluetech — APPROVED** [review]

File: changelog/9396.bugfix.rst (hunk 3 above)
Comment (nicoddemus) [reply changelog/9396.bugfix.rst:1]: Good idea, done!

File: src/_pytest/legacypath.py [hunk 4]
```diff
@@ -401,26 +401,10 @@ def Node_fspath_set(self: Node, value: LEGACY_PATH) -> None:
 
 
 @pytest.hookimpl
```
Comment (RonnyPfannschmidt) [inline src/_pytest/legacypath.py:403]: please set tryfirst=True

Comment (nicoddemus) [reply src/_pytest/legacypath.py]: Done, thanks

File: testing/test_config.py [hunk 5]
```diff
@@ -1264,15 +1264,16 @@ def pytest_load_initial_conftests(self):
     m = My()
     pm.register(m)
     hc = pm.hook.pytest_load_initial_conftests
-    values = hc._nonwrappers + hc._wrappers
-    expected = [
+    assert [x.function.__module__ for x in hc._nonwrappers] == [
```
Comment (nicoddemus) [inline testing/test_config.py:1267]: Decided to also split checking wrappers and non-wrappers, seems better to explicitly test them separately.

**Review by RonnyPfannschmidt — APPROVED** [review]

**Comment by The-Compiler** [comment]
I've held back merging this as I wasn't sure if we preferred to revert the plugin entirely or not - with #9438 I assume we decided to keep it, but this conflicts now due to the import changes. After the conflicts are resolved I think this is good to go. Thanks @nicoddemus and happy new year! :tada:

**Comment by nicoddemus** [comment]
I will fix this later today or tomorrow the latest! Happy new year!

**Pull request merged**